src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -141,14 +141,14 @@ class ClusterMonitorActor(val cluster: Cluster,
       // Remove credentials from instance metadata.
       // Only happens if an notebook service account was used.
       _ <- if (clusterStatus == ClusterStatus.Creating) removeCredentialsFromMetadata else Future.successful(())
-      // Ensure the cluster is ready for proxying but updating the IP -> DNS cache
-      _ <- ensureClusterReadyForProxying(publicIp, clusterStatus)
       // create or update instances in the DB
       _ <- persistInstances(instances)
       // update DB after auth futures finish
       _ <- dbRef.inTransaction { dataAccess =>
         dataAccess.clusterQuery.setToRunning(cluster.id, publicIp)
       }
+      // Ensure the cluster is ready for proxying by updating the IP -> DNS cache
+      _ <- ensureClusterReadyForProxying(publicIp, clusterStatus)
```

Comment:
Moved this after the DB update to prevent a race condition: 
1. DNS cache is update
2. DNS cache overwrites the new value with `ClusterNotReady` because the DB is not updated yet
3. DB is finally updated

I think this may be resolved with Vikram's work on the Dns cache anyway, but this may be a good short term fix to the tests.